frontend/src/zodTypes.ts

```diff
@@ -84,19 +112,19 @@ export type TankMetaDataRequest = ValidatedRequest<CreateTankMetaData>;
 
 export const serviceCallSchema = z.object({
   id: z.number().int(),
-  isApproved: z.boolean().optional(),
+  isApproved: z.boolean(),
   createdOn: z.coerce.date(),
 
-  customerRequest: z.string().optional(),
-  employeeNotes: z.string().optional(),
+  customerRequest: z.string().nullable().default(null),
+  employeeNotes: z.string().nullable().default(null),
   // server use only for not-approved notes
-  notApprovedNotes: z.string().optional(),
-  notesUpdated: z.coerce.date().optional(),
+  notApprovedNotes: z.string().nullable().default(null),
+  notesUpdated: z.coerce.date().nullable().default(null),
 
-  alkalinity: z.coerce.number(),
-  calcium: z.coerce.number(),
-  nitrate: z.coerce.number(),
-  phosphate: z.coerce.number(),
+  alkalinity: z.number(),
```

Comment:
the coerce for these four fields were lost

Reply:
we should rethink if we need coerce here. it might be best to get to the root of the problem and have them saved as number in the state

Reply:
The textfields handle input with strings. Parsing during the onChange function would be a way to convert it to a number then validate.